Marlin/src/lcd/ultralcd.cpp

```diff
@@ -2997,7 +3000,7 @@ void kill_screen(const char* lcd_msg) {
     #endif
     MENU_ITEM(function, MSG_RESTORE_FAILSAFE, lcd_factory_settings);
     #if ENABLED(EEPROM_SETTINGS)
-      MENU_ITEM(gcode, MSG_INIT_EEPROM, PSTR("M502\nM500")); // TODO: Add "Are You Sure?" step
+      MENU_ITEM(gcode, MSG_INIT_EEPROM, PSTR("M502\nM500\nM501")); // TODO: Add "Are You Sure?" step
```

Comment:
If `M501` is required, then there must be a bug in `M502`. It makes no sense to simply read back what was just written.

Reply:
M501 has always been required to get correct initialization.   That doesn't make sense, but historically, that is true.  I think it does indicate a bug in M502 or M500.

Reply:
With so many changes to the codebase, including bug-fixes to the EEPROM code over the last several months, the historical kludge may no longer be needed. Both `M502` and `M501` write into the same memory slots, and both call the same (new-ish) method to recalculate derived values based on settings. I'll do a careful audit and see whether `M501` does anything special compared to `M502`.